server/src/main/java/org/opensearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -996,50 +997,75 @@ static Settings aggregateIndexSettings(
     private static void updateReplicationStrategy(
         Settings.Builder settingsBuilder,
         Settings requestSettings,
-        Settings clusterSettings,
-        Settings combinedTemplateSettings
+        Settings nodeSettings,
+        Settings combinedTemplateSettings,
+        ClusterSettings clusterSettings
     ) {
         // The replication setting is applied in the following order:
         // 1. Explicit index creation request parameter
         // 2. Template property for replication type
-        // 3. Defaults to segment if remote store attributes on the cluster
+        // 3. Defaults to segment if remote store attributes/migration on the cluster
         // 4. Default cluster level setting
 
         final ReplicationType indexReplicationType;
         if (INDEX_REPLICATION_TYPE_SETTING.exists(requestSettings)) {
             indexReplicationType = INDEX_REPLICATION_TYPE_SETTING.get(requestSettings);
         } else if (INDEX_REPLICATION_TYPE_SETTING.exists(combinedTemplateSettings)) {
             indexReplicationType = INDEX_REPLICATION_TYPE_SETTING.get(combinedTemplateSettings);
-        } else if (CLUSTER_REPLICATION_TYPE_SETTING.exists(clusterSettings)) {
-            indexReplicationType = CLUSTER_REPLICATION_TYPE_SETTING.get(clusterSettings);
-        } else if (isRemoteDataAttributePresent(clusterSettings)) {
+        } else if (CLUSTER_REPLICATION_TYPE_SETTING.exists(nodeSettings)) {
```

Comment:
this should check clusterSettings IMO. I think earlier we were considering nodeSettings as clusterSettings which was incorrect.

Reply:
lets keep using nodesettings to be in parity with previous code. If needed create a separate task to address this